_config/security.yml

```diff
@@ -1,4 +1,33 @@
-SilverStripe\Security\MemberLoginForm:
+---
+Name: coresecurity
+---
+SilverStripe\Security\MemberAuthenticator\LoginForm:
   required_fields:
     - Email
-    - Password
\ No newline at end of file
+    - Password
+
+SilverStripe\Core\Injector\Injector:
+  SilverStripe\Control\RequestProcessor:
+    properties:
+      filters:
+        - %$SilverStripe\Security\AuthenticationRequestFilter
+  SilverStripe\Security\MemberAuthenticator\SessionAuthenticationHandler:
+    properties:
+      SessionVariable: loggedInAs
+  SilverStripe\Security\MemberAuthenticator\CookieAuthenticationHandler:
+    properties:
+      TokenCookieName: alc_enc
+      DeviceCookieName: alc_device
+      CascadeLogInTo: %$SilverStripe\Security\MemberAuthenticator\SessionAuthenticationHandler
+  SilverStripe\Security\IdentityStore:
+    class: SilverStripe\Security\AuthenticationRequestFilter
+  SilverStripe\Security\Security:
+    properties:
+      authenticators:
+        default: %$SilverStripe\Security\MemberAuthenticator\Authenticator
```

Comment:
tractorcow 2 days ago Contributor

As I commented on SilverStripe\Security\AuthenticationRequestFilter, please shift these from injecting service names to injecting the services themselves directly.

setup getAuthenticators() / setAuthenticators() on Security class, and assign an array of named handlers.

My suggestion is that you make these basic getters / setters on a protected $authenticators property, and rename the current getAuthenticators($service) to getApplicableAuthenticators($service).
Firesphere 2 days ago Contributor

That would disable the option to override the default authenticator. If I implement my own authenticator and want to override, I have to awkwardly use the YML, where as

authenticators:
  default: My\Authenticator\Class

Makes more sense to me. As this would also leave the option of a custom implementation, to fall back to the default authenticator (say I lost my FB/Twitter login, 2FA thing, etc.)
tractorcow a day ago Contributor

No, it wouldn't. you can key your array just as you have above, and you can override it with a like yml config block. All that you are changing is moving from injecting a class name to a class instance, and the config is moved up one level to the Injector top level key.